### ISSUE TITLE
packit-stable should follow Fedora branching

### ISSUE THREAD

**Issue by TomasTomecek** [packit/packit-service#1753]
Context: https://sentry.io/share/issue/b39eb764362649ca97a6704b125bd0db/

We NEED to make sure that packit-stable follows Fedora branching aggressively. As soon as the new Fedora release is available in Copr, we should have in our projects ASAP.

Next branching happens first week of Feb 2023.

_Edit_: date: 7th February

**Comment by praiskup** [comment]
JFTR, there's `$ copr list-chroots` api, not sure if related.

**Comment by lachmanfrantisek** [comment]
It should have been fixed by the work Maťo has done (switching to dnf for installations of Packit during the SRPM build).

Let's check during the branching.

**Comment by TomasTomecek** [comment]
I can confirm this is properly resolved: @praiskup and @frostyx confirmed that upon branching, fedora-latest within Copr changes to the newly branched version, but... we don't care about that because we always specify the latest stable release :) @mfocko fixed it properly

relevant commits:
https://github.com/packit/packit-service/commit/0ea24ed30da31f29b3874319f244aa320ecbc730
https://github.com/packit/packit-service/commit/7794101fd24d95a1558708c8ee30d560fc799743

https://github.com/packit/packit-service/blob/7794101fd24d95a1558708c8ee30d560fc799743/packit_service/worker/helpers/build/copr_build.py#L518-L520

if anything goes wrong after next week branching, let's track it as a new issue

**Comment by mfocko** [comment]
IIRC Copr doesn't rebuild during the branching, so we should probably make sure that the build is there

**Comment by TomasTomecek** [comment]
I understand that with the current code, branching should have no effect on our SRPM builds.

**Comment by mfocko** [comment]
tl;dr irrelevant, I thought we build from latest »branched« or rawhide /o\

Sorry, I should've been more explicit :D

We install from `packit/packit-{dev,stable}` Copr repo and we build in those repos when pushing to `main` (or `stable` respectively). That triggers builds to the `fedora-all` (or `fedora-{stable,rawhide}`; not sure, doesn't really matter).

My concern lies in the fact that »Copr doesn't rebuild when chroot is added«, therefore if they branch out and bump everything, we might not have a Packit build ready, cause we have older chroots.

---

And it's totally irrelevant, cause in the snippet in your comment above, I see `latest_fedora_stable_chroot`, so I guess the only issue would be:

1. brand new chroot appears
2. no build of packit happens
3. gets bumped as stable

**Comment by TomasTomecek** [comment]
> And it's totally irrelevant, cause in the snippet in your comment above, I see `latest_fedora_stable_chroot`

that's exactly my point: we won't use the newly branched Fedora in the SRPM build process. 
Upon the next branching, we will still use F37

> so I guess the only issue would be:
> 
> 1. brand new chroot appears
> 2. no build of packit happens
> 3. gets bumped as stable

yup, that's what I hope will happen: once we push to stable, we get builds for the new fedora, plenty of time before the next Fedora version gets officially released. Everything is awesome

but given how convoluted this is, I can totally imagine we are missing something and will have a blast on 7th :)
